components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/impl/UserInforRequestDefaultValidator.java

```diff
@@ -95,11 +96,34 @@ public String validateRequest(HttpServletRequest request) throws UserInfoEndpoin
                         "Content-Type header is wrong");
             }
         }
+
         String[] authzHeaderInfo = authzHeaders.trim().split(" ");
-        if (authzHeaderInfo.length < 2 || !BEARER.equals(authzHeaderInfo[0])) {
-            throw new UserInfoEndpointException(OAuthError.ResourceResponse.INVALID_REQUEST, "Bearer token missing");
+
+        if (authzHeaderInfo.length < 2) {
+            throw new UserInfoEndpointException(
+                    OAuthError.ResourceResponse.INVALID_REQUEST, "Token missing"
+            );
+        }
+
+        String authScheme = authzHeaderInfo[0];
+
+        if (BEARER.equals(authScheme)) {
+            // Bearer token. For a Bearer token no additional DPoP header is expected.
+            return authzHeaderInfo[1];
+        } else if (DPOP.equals(authScheme)) {
+            // DPoP token: the request MUST include a DPoP header
+            String dpopHeader = request.getHeader(DPOP);
+            if (dpopHeader == null || dpopHeader.trim().isEmpty()) {
+                throw new UserInfoEndpointException(
+                        OAuthError.ResourceResponse.INVALID_REQUEST, "DPoP header is required with DPoP tokens"
+                );
+            }
+            return authzHeaderInfo[1];
+        } else {
+            throw new UserInfoEndpointException(
```

Comment:
Have we changed the error msgs now?

Reply:
Old error messages were preserved for backward compatibility through https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2811/commits/0a551dd44ad66476207dae580dab1a8948bc236e.